### ISSUE TITLE
Baggage not propagated for consumers who use ruby_kafka instrumentation

### ISSUE THREAD

**Issue by Asafb26** [open-telemetry/opentelemetry-ruby#549]
currently, there is a different logic between:
```ruby
parent_context = OpenTelemetry.propagation.text.extract({traceparent: ..., baggage: my_baggage})
tracer.in_span("test", with_parent: parent_context, attributes: attributes, kind: :consumer) do
  puts OpenTelemetry.baggage.values
end

#prints nil
```
and
```ruby
parent_context = OpenTelemetry.propagation.text.extract({traceparent: ..., baggage: my_baggage})
OpenTelemetry::Context.with_current(parent_context) do
  tracer.in_span("test", attributes: attributes, kind: :consumer) do
    puts OpenTelemetry.baggage.values
  end
end

#prints the content of my_baggage
```

The first keeps the current context active while use the parent span_context data, while the second replaces the current context, yet use the parent span_context data (from the replaced context).

The issue starts with the fact that baggage is not a part of span_context, so for proper propagation of baggage data - we must use the second approach.

**Comment by fbogsany** [comment]
This difference seems really unfortunate. Are we correctly implementing the spec here in `tracer.in_span(..., with_parent: ...)`? If not, we should fix that. If we are, I wonder if this difference in behaviour should be raised as a spec issue?

**Comment by Asafb26** [comment]
AFAIK most instrumentation uses the second approach, so in my perspective - if it's not an issue, I can align the implementation of ruby_kafka with the rest.
regarding the specs - the baggage is not a part of the span_context, which I found it bit weird, but yet confirms that we are aligned to the specs

**Comment by fbogsany** [comment]
Fixed in #550
